src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListServiceImpl.java

```diff
@@ -81,6 +150,116 @@ public GermplasmListGeneratorDTO create(final GermplasmListGeneratorDTO request,
+
+		final List<GermplasmListData> idGermplasmListDataSaved = new ArrayList<>();
+		try {
+			final List<Integer> deletedListEntryIds = new ArrayList<>();
+			data.forEach(germplasmListData -> {
+				String groupName = germplasmListData.getGroupName();
+				if (groupName.length() > MAX_CROSS_NAME_SIZE) {
+					groupName = groupName.substring(0, MAX_CROSS_NAME_SIZE - 1);
+					groupName = groupName + TRUNCATED;
+					germplasmListData.setGroupName(groupName);
+				}
+
+				final GermplasmListData recordSaved = this.daoFactory.getGermplasmListDataDAO().saveOrUpdate(germplasmListData);
+				idGermplasmListDataSaved.add(recordSaved);
+				if (!Objects.isNull(germplasmListData.getStatus()) && germplasmListData.getStatus() == 9) {
+					deletedListEntryIds.add(germplasmListData.getId());
+				}
+			});
+
+			if (!deletedListEntryIds.isEmpty()) {
+				this.daoFactory.getTransactionDAO().cancelUnconfirmedTransactionsForListEntries(deletedListEntryIds);
+			}
+
+		} catch (final Exception e) {
+
+			throw new MiddlewareQueryException(
+				"Error encountered while saving Germplasm List Data: GermplasmListManager.addOrUpdateGermplasmListData(data="
+					+ data + "): " + e.getMessage(),
+				e);
+		}
+
+		return idGermplasmListDataSaved;
+	}
+
+	@Override
+	public void addGermplasmEntriesToList(final Integer germplasmListId,
+		final SearchCompositeDto<GermplasmSearchRequest, Integer> searchComposite, final String programUUID) {
+
+		final GermplasmList germplasmList = this.getGermplasmListById(germplasmListId)
+			.orElseThrow(() -> new MiddlewareRequestException("", "list.not.found"));
+
+		//Get the germplasm entries to add
+		final List<AddGermplasmEntryModel> addGermplasmEntriesModels = new ArrayList<>();
+		if (CollectionUtils.isEmpty(searchComposite.getItemIds())) {
+			final GermplasmSearchRequest searchRequest = searchComposite.getSearchRequest();
+			if(Objects.isNull(searchRequest.getAddedColumnsPropertyIds())) {
+				searchRequest.setAddedColumnsPropertyIds(new ArrayList<>());
+			}
+			if (!searchRequest.getAddedColumnsPropertyIds().contains(GERMPLASM_PREFERRED_NAME)) {
+				searchRequest.getAddedColumnsPropertyIds().add(GERMPLASM_PREFERRED_NAME);
+			}
+
+			this.germplasmSearchService.searchGermplasm(searchRequest, null, programUUID)
+				.forEach(germplasmSearchResponse -> addGermplasmEntriesModels.add(new AddGermplasmEntryModel(
+					germplasmSearchResponse.getGid(),
+					germplasmSearchResponse.getGermplasmPeferredName(),
+					germplasmSearchResponse.getGroupId()
+				)));
+
+		} else {
+
+			this.germplasmService.getGermplasmByGIDs(new ArrayList<>(searchComposite.getItemIds()))
+				.forEach(germplasm -> addGermplasmEntriesModels.add(new AddGermplasmEntryModel(
+						germplasm.getGid(),
+						germplasm.getPreferredName().getNval(),
+						germplasm.getMgid())
+					)
+				);
+		}
+
+		//Get the entryId max value
+		final int maxEntryId = germplasmList.getListData()
```

Comment:
For business purposes I think entryNo work better than entryId, not sure why the called the attribute entryId. Could you please rename the variables to use entryNo instead of entryId ? @lik-83